lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -0,0 +1,3045 @@
+  # with keys area_ft2, type, fuel, and zones (an array of zones)
+  def get_baseline_system_groups_for_one_building_type(model, hvac_building_type, zones_in_building_type)
+    # Build zones hash of [zone, zone area, occupancy type, building type, fuel]
+    zones = model_zones_with_occ_and_fuel_type(model, 'custom')
+
+    # Ensure that there is at least one conditioned zone
+    if zones.size.zero?
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Model', 'The building does not appear to have any conditioned zones. Make sure zones have thermostat with appropriate heating and cooling setpoint schedules.')
+      return []
+    end
+
+    # Consider special rules for computer rooms
+    # need load of all
+
+    # Get cooling load of all computer rooms to establish system types
+    comp_room_loads = {}
+    bldg_comp_room_load = 0
+    zones.each do |zn|
+      zone_load = 0.0
+      has_computer_room = false
+      # First check if any space in zone has a computer room
+      zn['zone'].spaces.each do |space|
+        if space.spaceType.get.standardsSpaceType.get == 'computer room'
+          has_computer_room = true
+          break
+        end
+      end
+      if has_computer_room
+        # Collect load for entire zone
+        zone_load_w = zn['zone'].coolingDesignLoad.to_f
+        zone_load_w *= zn['zone'].floorArea * zn['zone'].multiplier
+        zone_load = OpenStudio.convert(zone_load_w, 'W', 'Btu/hr').get
+      end
+      comp_room_loads[zn['zone'].name.get] = zone_load
+      bldg_comp_room_load += zone_load
+    end
+
+    # Lab zones are grouped separately if total lab exhaust in building > 15000 cfm
+    # Make list of zone objects that contain laboratory spaces
+    lab_zones = []
+    has_lab_spaces = {}
+    model.getThermalZones.sort.each do |zone|
+      # Check if this zone includes laboratory space
+      zone.spaces.each do |space|
+        spacetype = space.spaceType.get.standardsSpaceType.get
+        has_lab_spaces[zone.name.get] = false
+        if space.spaceType.get.standardsSpaceType.get == 'laboratory'
+          lab_zones << zone
+          has_lab_spaces[zone.name.get] = true
+          break
+        end
+      end
+    end
+
+    lab_exhaust_si = 0
+    lab_relief_si = 0
+    if !lab_zones.empty?
+      # Build a hash of return_node:zone_name
+      node_list = {}
+      zone_return_flow_si = Hash.new(0)
+      var_name = 'System Node Standard Density Volume Flow Rate'
+      frequency = 'hourly'
+      model.getThermalZones.each do |zone|
+        port_list = zone.returnPortList
+        port_list_objects = port_list.modelObjects
+        port_list_objects.each do |node|
+          node_name = node.nameString
+          node_list[node_name] = zone.name.get
+        end
+        zone_return_flow_si[zone.name.get] = 0
+      end
+
+      # Get return air flow for each zone (even non-lab zones are needed)
+      # Take from hourly reports created during sizing run
```

Comment:
need to do a sizing run here if one wasn't done already

Reply:
@mdahlhausen : I think the sizing run done here: 
https://github.com/NREL/openstudio-standards/blob/ca24c63850bc421a376011e01ba363271f7a41c3/lib/openstudio-standards/standards/Standards.Model.rb#L137
is acceptable, since the lab exhaust does not change between the user model and the baseline.